rest/permissions.py

```diff
@@ -130,3 +134,69 @@ def has_permission(self, request, view):
             return True
         return RepoPermission.manage_repo_users[0] in get_perms(
             request.user, repo)
+
+
+class AddEditMetadataPermission(BasePermission):
+    """Checks add_edit_metadata permission"""
+
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+
+        try:
+            repo = get_repo(view.kwargs['repo_slug'], request.user.id)
+        except NotFound:
+            raise Http404()
+        except PermissionDenied:
+            return False
+
+        return (
+            RepoPermission.add_edit_metadata[0]
+            in get_perms(request.user, repo)
+        )
+
+
+class ViewLearningResourcePermission(ViewRepoPermission):
+    """
+    Checks that user has view_repo permission for the repository
+    and that that repo owns the learning resource being requested
+    """
+
+    def has_permission(self, request, view):
+        if not super(ViewLearningResourcePermission, self).has_permission(
+                request, view):
+            return False
+
+        repo_slug = view.kwargs['repo_slug']
+        lr_id = int(view.kwargs['lr_id'])
+
+        try:
+            LearningResource.objects.filter(
+                course__repository__slug=repo_slug).get(id=lr_id)
+        except LearningResource.DoesNotExist:
+            raise Http404()
+
+        return True
+
+
+class ViewStaticAssetPermission(ViewLearningResourcePermission):
+    """
+    Checks that user has view_repo permission for the repository
+    and that that repo owns the learning resource being requested
```

Comment:
nits:

"learning resource" here, but "LearningResource" in lots of other comments.

No period at the end.